src/main/java/com/programmers/springbootbasic/domain/voucher/Voucher.java

```diff
@@ -0,0 +1,85 @@
+package com.programmers.springbootbasic.domain.voucher;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+public abstract class Voucher {
+    private final UUID voucherId;
+    private final String name;
+    private final Long minimumPriceCondition;
+    private final LocalDateTime expirationDate;
+    private VoucherState state = VoucherState.AVAILABLE;
+
+    protected Voucher(UUID voucherId, String name, LocalDateTime expirationDate) {
+        if (isInvalidVoucherId(voucherId)) {
+            throw new IllegalArgumentException("잘못된 아이디 값");
+        }
+        this.voucherId = voucherId;
+        if (isInvalidName(name)) {
+            throw new IllegalArgumentException("잘못된 이름");
+        }
+        this.name = name;
+        this.minimumPriceCondition = 0L;
+        if (isInvalidExpirationDate(expirationDate)) {
+            throw new IllegalArgumentException("잘못된 유효기간");
+        }
+        this.expirationDate = expirationDate;
+    }
+
+    protected Voucher(UUID voucherId, String name, Long minimumPriceCondition, LocalDateTime expirationDate) {
+        if (isInvalidVoucherId(voucherId)) {
+            throw new IllegalArgumentException("잘못된 아이디 값");
+        }
+        this.voucherId = voucherId;
+        if (isInvalidName(name)) {
+            throw new IllegalArgumentException("잘못된 이름");
+        }
+        this.name = name;
+        this.minimumPriceCondition = minimumPriceCondition;
+        if (isInvalidExpirationDate(expirationDate)) {
+            throw new IllegalArgumentException("잘못된 유효기간");
+        }
+        this.expirationDate = expirationDate;
+    }
```

Comment:
```suggestion
    protected Voucher(UUID voucherId, String name, LocalDateTime expirationDate) {
        this(voucherId, name, null, expirationDate);
    }

    protected Voucher(UUID voucherId, String name, Long minimumPriceCondition, LocalDateTime expirationDate) {
        if (isInvalidVoucherId(voucherId)) {
            throw new IllegalArgumentException("잘못된 아이디 값. voucherId=" + voucherId);
        }
        this.voucherId = voucherId;
        if (isInvalidName(name)) {
            throw new IllegalArgumentException("잘못된 이름. name="+name);
        }
        this.name = name;
        this.minimumPriceCondition = minimumPriceCondition;
        if (isInvalidExpirationDate(expirationDate)) {
            throw new IllegalArgumentException("잘못된 유효기간. expirationDate="+expirationDate);
        }
        this.expirationDate = expirationDate;
    }
```
추가적으로 이렇게 작성해주면 코드 중복이 줄어들 것 같네요.
그리고 예외 메시지를 구체적으로 작성해주면 좋을 것 같아요.

Reply:
생성자 중복부분, 예외메세지 말씀 주신 방법으로 변경했습니다.


[22d2a24](https://github.com/prgrms-be-devcourse/springboot-basic/pull/660/commits/22d2a24ce11d4b2599d018304b86b4b14a6f3434)